#!/usr/bin/env bash

#PBS -N fastp_trim_qc
#PBS -l nodes=1:ppn=20 #1 of node requested; 20 processors in total
#PBS -l walltime=40:00:00 # 72 hours walltime
#PBS -l mem=100gb #100 Gb memory in total
#PBS -m abe
#PBS -M MohammedHisham.Shaikh@ugent.be
#PBS -o /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_output.log  
#PBS -e /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_error.log   



#Making sure I am on the rigth cluster
module swap cluster/doduo


# Unset OMP_PROC_BIND
unset OMP_PROC_BIND



# AIM: To assess the initial quality of raw reads from NJ200, PE cruises and GF2018 and trim them

. ~/vo_vliz.sh
. .bashrc

# Defining directories
HOME_DIR="/data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics"
OUTPUT_DIR="$HOME_DIR/working/1_read_qc_trimming/1_fastp_trim_qc"
SAMPLE_DIR="$HOME_DIR/data/raw_reads"
SAMPLE_LIST_NJ_PE="$HOME_DIR/data/sample_overview/NJ_PE_sample_list.csv"
SAMPLE_LIST_GF="$HOME_DIR/data/sample_overview/GF_sample_list.csv"


mkdir -p $OUTPUT_DIR
cd $OUTPUT_DIR

# Activate fastp
mamba activate fastp

# Running fastp on NJ2020 and PE samples 
for sample in $(cat $SAMPLE_LIST_NJ_PE); do
    echo "Processing sample: $sample"
    fastp -i "$SAMPLE_DIR/${sample}_R1.fastq.gz" -I "$SAMPLE_DIR/${sample}_R2.fastq.gz" -o ${sample}_R1_paired.fastq.gz -O ${sample}_R2_paired.fastq.gz --unpaired1 ${sample}_R1_unpaired.fastq.gz --unpaired2 ${sample}_R2_unpaired.fastq.gz --html "${sample}_fastp.html" --json "${sample}_fastp.json" --report_title "${sample} fastp report" --thread 16 --compression 4 --verbose --dedup --dup_calc_accuracy 6 --detect_adapter_for_pe --trim_front1 15 --trim_front2 15 --trim_tail1 15 --trim_tail2 15 --cut_right --cut_front --cut_mean_quality 30 --qualified_quality_phred 33 --length_required 50 --correction --overrepresentation_analysis
done




# Running fastp on GF samples 
for sample in $(cat $SAMPLE_LIST_GF); do
    echo "Processing sample: $sample"
    fastp -i "$SAMPLE_DIR/${sample}_R1.fastq.gz" -I "$SAMPLE_DIR/${sample}_R2.fastq.gz" -o ${sample}_R1_paired.fastq.gz -O ${sample}_R2_paired.fastq.gz --unpaired1 ${sample}_R1_unpaired.fastq.gz --unpaired2 ${sample}_R2_unpaired.fastq.gz --html "${sample}_fastp.html" --json "${sample}_fastp.json" --report_title "${sample} fastp report" --thread 16 --compression 4 --verbose --dedup --dup_calc_accuracy 6 --detect_adapter_for_pe --trim_tail1 50 --trim_tail2 50 --cut_right --cut_front --cut_mean_quality 30 --qualified_quality_phred 33 --length_required 50 --correction --overrepresentation_analysis
done


mamba deactivate


mamba activate multiqc

multiqc -d . -o ./multiqc -n trimmed_reads_multiqc_report.html


mamba deactivate