#!/usr/bin/env bash

#PBS -N genomad_individual_assemblies
#PBS -l nodes=1:ppn=40 #1 of node requested; 20 processors in total
#PBS -l walltime=72:00:00 # 48 hours walltime
#PBS -l mem=200gb #150 Gb memory in total
#PBS -m abe
#PBS -M MohammedHisham.Shaikh@ugent.be
#PBS -o /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_output.log  
#PBS -e /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_error.log   



#Making sure I am on the rigth cluster
module swap cluster/doduo

# Unset OMP_PROC_BIND
unset OMP_PROC_BIND


# AIM: To classify viral contigs

. ~/vo_vliz.sh
. .bashrc

# Defining directories
HOME_DIR="/data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics"
OUTPUT_DIR="$HOME_DIR/working/4_genomad/1_genomad_individual_assemblies"
ASSEMBLY_DIR="$HOME_DIR/working/2_assembly/1_individual_assemblies"
SAMPLE_LIST_NJ_PE_GF="$HOME_DIR/data/sample_overview/NJ_PE_GF_sample_list.csv"
GENOMAD_DB="$HOME/viral_tools/genomad/genomad_db"


mkdir -p $OUTPUT_DIR
cd $OUTPUT_DIR


mamba activate genomad


for sample in $(cat $SAMPLE_LIST_NJ_PE_GF); do
  
    echo "Running Genomad end-to-end for sample: $sample"

    # Running genomad
    genomad end-to-end --threads 40 --cleanup --splits 16 "$ASSEMBLY_DIR/${sample}_megahit_assembly/${sample}_final_contigs.fa" "$OUTPUT_DIR/${sample}_genomad_output" "$GENOMAD_DB"
 
done 


mamba deactivate