#!/usr/bin/env bash

#PBS -N megahit_individual_assemblies
#PBS -l nodes=1:ppn=40 #1 of node requested; 20 processors in total
#PBS -l walltime=48:00:00 # 48 hours walltime
#PBS -l mem=150gb #150 Gb memory in total
#PBS -m abe
#PBS -M MohammedHisham.Shaikh@ugent.be
#PBS -o /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_output.log  
#PBS -e /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_error.log   



#Making sure I am on the rigth cluster
module swap cluster/doduo

# Unset OMP_PROC_BIND
unset OMP_PROC_BIND



# AIM: To assemble individual MEGAHIT assemblies, assess the qualityusing metaQUAST and compile the report using multiQC

. ~/vo_vliz.sh
. .bashrc

# Defining directories
HOME_DIR="/data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics"
OUTPUT_DIR="$HOME_DIR/working/2_assembly/1_individual_assemblies"
TRIMMED_READS_DIR="$HOME_DIR/working/1_read_qc_trimming/1_fastp_trim_qc"
SAMPLE_LIST_NJ_PE_GF="$HOME_DIR/data/sample_overview/NJ_PE_GF_sample_list.csv"
METAQUAST_DIR="$OUTPUT_DIR/../3_metaquast_individual_assemblies"

mkdir -p $OUTPUT_DIR $METAQUAST_DIR
cd $OUTPUT_DIR


# Activate MEGAHIT



# Running MEGAHIT 
for sample in $(cat $SAMPLE_LIST_NJ_PE_GF); do
    echo "Processing sample: $sample"
    
    # Step 1: Running megahit
    mamba activate megahit

    megahit -1 $TRIMMED_READS_DIR/${sample}_R1_paired.fastq.gz -2 $TRIMMED_READS_DIR/${sample}_R2_paired.fastq.gz -o "$OUTPUT_DIR/${sample}_megahit_assembly" -t 40 -m 0.8

    mamba deactivate

# renaming the output file

  if [ -f "$OUTPUT_DIR/${sample}_megahit_assembly/final.contigs.fa" ]; then
        mv "$OUTPUT_DIR/${sample}_megahit_assembly/final.contigs.fa" "$OUTPUT_DIR/${sample}_megahit_assembly/${sample}_final_contigs.fa"
        echo "Renamed final.contigs.fa to ${sample}_final_contigs.fa"
    else
        echo "Warning: final.contigs.fa not found for $sample"
    fi

    # Step 2: Running metaQUAST
    echo "Running MetaQUAST for $sample"

    mamba activate quast

    metaquast.py "$OUTPUT_DIR/${sample}_megahit_assembly/${sample}_final_contigs.fa" \
                 -o "$METAQUAST_DIR/${sample}_metaquast" \
                 --max-ref-number 0 \
                 --threads 40

    mamba deactivate

done 


mamba activate multiqc

multiqc $METAQUAST_DIR -o $METAQUAST_DIR/multiqc -n individual_assembly_metaquast_multiqc_report.html

mamba deactivate


