#!/usr/bin/env bash

#PBS -N bowtie2_genomad_mapping_quant_NJ_PE_GF_clustered
#PBS -l nodes=1:ppn=10 #1 of node requested; 20 processors in total
#PBS -l walltime=24:00:00 # 48 hours walltime
#PBS -l mem=100gb #150 Gb memory in total
#PBS -m abe
#PBS -M MohammedHisham.Shaikh@ugent.be
#PBS -o /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_output.log  
#PBS -e /data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics/logs/${PBS_JOBNAME}_error.log   



#Making sure I am on the rigth cluster
module swap cluster/doduo

# Unset OMP_PROC_BIND
unset OMP_PROC_BIND



# AIM: Read mapping and quantifictaion for genomad viruses in NJ_PE_GF clustered ross_assembly

# Read mapping is required to get depth coverage necessary for binning and also for read quanitfication

. ~/vo_vliz.sh
. .bashrc

# Defining directories
HOME_DIR="/data/gent/vo/001/gvo00125/vsc44392/Projects/North_Sea_Viral_Metagenomics"
OUTPUT_DIR="$HOME_DIR/working/5_read_mapping/3_NJ_PE_GF_genomad"
TRIMMED_READS_DIR="$HOME_DIR/working/1_read_qc_trimming/1_fastp_trim_qc"
GENOMAD_FASTA="$HOME_DIR/working/4_genomad/2_genomad_cross_assemblies/NJ_PE_GF_genomad_output/NJ_PE_GF_clustered_contigs_summary/NJ_PE_GF_clustered_contigs_virus.fna"
SAMPLE_LIST="$HOME_DIR/data/sample_overview/NJ_PE_GF_sample_list.csv"
INDEX_DIR="$OUTPUT_DIR/index"
MAPPING_DIR="$OUTPUT_DIR/mapping"
COMBINED_QUANT_DIR="$OUTPUT_DIR/combined_quant"

mkdir -p $OUTPUT_DIR $INDEX_DIR $MAPPING_DIR $COMBINED_QUANT_DIR 
cd $OUTPUT_DIR



mamba activate bowtie2

# Building Bowtie2 index for the assembled contigs
bowtie2-build $GENOMAD_FASTA $INDEX_DIR/NJ_PE_GF_index

# Creating combined quantification file
COMBINED_QUANT="$COMBINED_QUANT_DIR/combined_read_counts_NJ_PE_GF.txt"

echo -e "Sample\tContig\tContig_Length\tMapped_Reads\tUnmapped_Reads" > $COMBINED_QUANT

# Loop through samples
for sample in $(cat $SAMPLE_LIST); do
    echo "Processing sample: $sample"
    
    # Define file paths for paired-end reads
    R1="$TRIMMED_READS_DIR/${sample}_R1_paired.fastq.gz"
    R2="$TRIMMED_READS_DIR/${sample}_R2_paired.fastq.gz"
    BAM="$MAPPING_DIR/${sample}_mapped_reads_sorted.bam"
    QUANT="$MAPPING_DIR/${sample}_read_counts.txt"

    # Map reads to contigs and generate BAM file
    bowtie2 -x $INDEX_DIR/NJ_PE_GF_index -1 $R1 -2 $R2 --threads 10 | samtools view -bS - | samtools sort -o $BAM

    # Indexing BAM file 
    samtools index $BAM

    # Quantifing reads mapped to contigs
    samtools idxstats $BAM > $QUANT

    # Adding sample name and aggregate into combined file
    awk -v sample="$sample" 'BEGIN {FS="\t"} {print sample"\t"$1"\t"$2"\t"$3"\t"$4}' $QUANT >> $COMBINED_QUANT

done 

echo "Quantification  completed for all samples. Combined results saved in $COMBINED_QUANT"

mamba deactivate